src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -131,21 +140,71 @@ object ModelJsonProtocol {
         case None => None
         case _ => throw DeserializationException("unexpected json type for " + SIZE)
       }
-      LibrarySearchParams(term, from, size)
+      val fields: Map[String, Seq[String]] = data.getOrElse(SEARCH_FIELDS, None) match {
+        case some:JsObject => some.convertTo[Map[String,Seq[String]]]
+        case None => Map.empty
+      }
```

Comment:
this needs the `case _` ... you could theoretically have a `JsArray` instead of a `JsObject`.